Move job annotations right after name

matthyx opened the pull request:
Fixes #13413

matthyx commented:
@spiffxp I have done the big bang...
If you are interested, I did some helper scripts in python to reorder map items, and validate yaml consistency.

matthyx commented:
(stupid comments, I will fix them tomorrow by hand)

neolit123 commented:
perhaps there should be an enforced formatting check in CI that backs up this change, otherwise this can drift again in the future.

matthyx commented:
> perhaps there should be an enforced formatting check in CI that backs up this change, otherwise this can drift again in the future.

Agreed, but I don't think yamllint supports it.

matthyx commented:
/assign @spiffxp

matthyx commented:
(I think it will be super painful to merge in one go...)

matthyx commented:
@spiffxp will you have some time to work on this next week?

spiffxp commented:
Jobs generated by config-forker and config-rotator have their fields alpha-sorted, I wouldn't bother updating them

I think you should add the scripts used to `experiment/`

Feel free to ping me on slack and I'll help move on this more quickly for you

matthyx commented:
@spiffxp I have added 2 scripts to experiments, and here is how you are supposed to use them.

`experiment/move-annotations.py config/jobs/gke/`
Will scan all files under this path and try to move annotations. Once run, the result should remain idempotent.

`experiment/compare-yaml.py config/jobs/gke/ /tmp/config/jobs/gke/`
Will compare the yaml deserialization of each file under `config/jobs/gke/` with those in `/tmp/config/jobs/gke/`. That second script is just to make sure I didn't screw anything.

On the second commit I have added the modifications on gke jobs as an example...

matthyx commented:
/test pull-test-infra-bazel

matthyx commented:
what's wrong with you `kettle`?

matthyx commented:
@spiffxp I think we're ready now... let me know where to start?

matthyx commented:
/remove-lifecycle stale

spiffxp commented on experiment/compare-yaml.py:
```diff
@@ -0,0 +1,59 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
```
Can you add a comment to the top of this file (or in main's docstring) explaining what this file does?

spiffxp commented on experiment/move-annotations.py:
```diff
@@ -0,0 +1,69 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
```
Can you add a comment to the top of this file (or in main's docstring) explaining what this file does?

matthyx commented:
@spiffxp I am thinking about including these as verify-annotations.sh and update-annotations.sh scripts, in order to include them as the bazel job checks... what do you think?

But to do so, I need to know which scripts should be checked, to avoid fighting with auto-generation.

spiffxp commented:
>@spiffxp I am thinking about including these as verify-annotations.sh and update-annotations.sh scripts, in order to include them as the bazel job checks... what do you think?

Sounds great

>But to do so, I need to know which scripts should be checked, to avoid fighting with auto-generation.

Not sure I understand this.  Which files to ignore because they're not completely hand-written?  I believe that would be files under:
- https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/generated (generated by https://github.com/kubernetes/test-infra/blob/master/releng/generate_tests.py)
- https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/sig-release/release-branch-jobs (generated by https://github.com/kubernetes/test-infra/tree/master/releng/config-forker)

matthyx commented:
/remove-lifecycle stale

spiffxp commented:
/test pull-test-infra-integration
#20498 (comment)

hasheddan commented:
@spiffxp is this ready to go at this point?

matthyx commented:
This PR only includes one script to relocate annotations within the job yaml files, and the second allow to compare 2 yaml files.

spiffxp commented:
https://github.com/kubernetes/test-infra/pull/16216#pullrequestreview-451978307 remains unaddressed

matthyx commented:
> [#16216 (review)](https://github.com/kubernetes/test-infra/pull/16216#pullrequestreview-451978307) remains unaddressed

sorry, will fix that

matthyx commented:
/remove-lifecycle stale

matthyx commented:
/remove-lifecycle rotten

spiffxp commented:
/test pull-test-infra-integration

spiffxp reviewed (approved):
/approve
/lgtm
